core/src/test/java/org/elasticsearch/index/percolator/ExtractQueryTermsServiceTests.java

```diff
@@ -218,6 +220,31 @@ public void testExtractQueryMetadata_boostQuery() {
         assertThat(terms.get(0).bytes(), equalTo(termQuery1.getTerm().bytes()));
     }
 
+    public void testExtractQueryMetadata_commonTermsQuery() {
```

Comment:
I'd love to see some high level testing on this. There is so much crazy stuff that goes on between the user's query text and the actual queries that it'd be nice to know that this works all the way from the API. Maybe some REST tests that the percolator still returns correct results for cross fields multi match queries?